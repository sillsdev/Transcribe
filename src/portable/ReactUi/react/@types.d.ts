declare module 'react-avatar';
declare module 'react-contextmenu';
declare module 'react-localization';
declare module 'react-player';
declare module 'react-string-replace';
declare module 'react-toggle-switch';
declare module 'react-tooltip';
declare module 'react-thunk';