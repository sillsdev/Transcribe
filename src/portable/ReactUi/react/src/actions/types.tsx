export const FETCH_USERS = 'FETCH_USERS';
export const SELECT_USER = 'SELECT_USER';
export const FETCH_TASKS = 'FETCH_TASKS';
export const SELECT_PROJECT = 'SELECT_PROJECT';
export const SELECT_TASK = 'SELECT_TASK';
export const ASSIGN_TASK_PENDING = 'ASSIGN_TASK_PENDING';
export const UNASSIGN_TASK_PENDING = 'UNASSIGN_TASK_PENDING';
export const PLAY_STATUS = 'PLAY_STATUS';
export const PLAYSPEEDRATE_CHANGE = 'PLAYSPEEDRATE_CHANGE';
export const JUMP_CHANGE = 'JUMPAHEAD_CHANGE';
export const FETCH_LOCALIZATION = 'FETCH_LOCALIZATION';
export const UPDATE_USER_PENDING = 'UPDATE_USER_PENDING';
export const WRITE_PENDING = 'WRITE_PENDING';
export const WRITE_FULFILLED = 'WRITE_FULFILLED';
export const SAVE_TOTAL_SECONDS = 'SAVE_TOTAL_SECONDS';
export const REPORT_POSITION = 'REPORT_POSITION';
export const REQUEST_POSITION = 'REQUEST_POSITION';
export const FETCH_TRANSCRIPTION = 'FETCH_TRANSCRIPTION';
export const INITIAL_TRANSCRIPTION = 'INITIAL_TRANSCRIPTION';
export const SAVE_STATUS = 'SAVE_STATUS';
export const SUBMIT_STATUS = 'SUBMIT_STATUS';
export const COMPLETE_TRANSCRIPTION_PENDING = 'COMPLETE_TRANSCRIPTION_PENDING';
export const COMPLETE_REVIEW_PENDING = 'COMPLETE_REVIEW_PENDING';
export const FETCH_PARATEXT_PROJECTS = 'FETCH_PARATEXT_PROJECTS';
export const SELECT_PARATEXT_PROJECT = 'SELECT_PARATEXT_PROJECT';
export const SELECT_POPUP_TASK = 'SELECT_POPUP_TASK';
export const UPDATE_TASK = 'UPDATE_TASK';
export const DELETE_TASK = 'DELETE_TASK';
export const SELECT_POPUP_USER = 'SELECT_POPUP_USER';
export const DELETE_USER = 'DELETE_USER';
export const UPDATE_PROJECT = 'UPDATE_PROJECT';
export const RESTORE_DEFAULT_USER_HOTKEYS = 'RESTORE_DEFAULT_USER_HOTKEYS';
export const COPY_TO_CLIPBOARD = 'COPY_TO_CLIPBOARD';
export const UPDATE_PROJECT_AVATAR = 'UPDATE_PROJECT_AVATAR';
export const CLEAR_SELECTED_PARATEXT_PROJECT = 'CLEAR_SELECTED_PARATEXT_PROJECT';
export const FETCH_ZTT_PROJECTS_COUNT = "FETCH_ZTT_PROJECTS_COUNT";
export const SAVE_AVATAR = "SAVE_AVATAR";
export const SET_AVATAR_RADIUS = 'SET_AVATAR_RADIUS';
export const SET_SAVE_TO_PROJECT = 'SET_SAVE_TO_PROJECT';
export const INIT_TASKS = 'INIT_TASKS';
export const ADD_MANY_TASKS = "ADD_MANY_TASKS";
export const SHOW_HELP = "SHOW HELP";
export const SET_PLAYED_SECONDS = "SET_PLAYED_SECONDS";
export const FETCH_META = "FETCH_META";
export const SET_SELECTED_OPTION = "SET_SELECTED_OPTION";
export const FETCH_FILTERED_TASK = 'FETCH_FILTERED_TASK';
export const SET_TODO_HIGHLIGHT = 'SET_TODO_HIGHLIGHT';