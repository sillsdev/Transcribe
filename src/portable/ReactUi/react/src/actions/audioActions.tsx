import Axios from 'axios';
import { log } from '../actions/logAction';
import { FETCH_META, INITIAL_TRANSCRIPTION, JUMP_CHANGE, PLAY_STATUS, PLAYSPEEDRATE_CHANGE,
    REPORT_POSITION, REQUEST_POSITION, SAVE_STATUS, SAVE_TOTAL_SECONDS, SET_PLAYED_SECONDS, SUBMIT_STATUS } from './types';

export function playStatus(playing: boolean): any{
    return {
        payload: playing,
        type: PLAY_STATUS
    }
}

export function playSpeedRateChange(playSpeedRate: number): any{
    return {
        payload: playSpeedRate,
        type: PLAYSPEEDRATE_CHANGE
    }
}

export function jumpChange(jump: number): any{
    return {
        payload: jump,
        type: JUMP_CHANGE
    }
}

export function saveTotalSeconds(seconds: number): any {
    return {
        payload: seconds,
        type: SAVE_TOTAL_SECONDS
    }
}

export const reportPosition = (taskid: string, seconds: number) => (dispatch: any) => {
    dispatch({payload: seconds, type: REPORT_POSITION});
    Axios.put('/api/ReportPosition?task=' + taskid + '&position=' + seconds.toString())
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + REPORT_POSITION + ", task=" + taskid + ", seconds=" + seconds))
        })
}

export function requestPosition(): any {
    return {
        type: REQUEST_POSITION
    }
}

export function setInitialTranscription(status: boolean): any {
    return {
        payload: status,
        type: INITIAL_TRANSCRIPTION
    }
}

export function setSubmitted(submit: boolean): any{
    return {
        payload: submit,
        type: SUBMIT_STATUS
    }
}

export function setSaved(saved: boolean): any{
    return {
        payload: saved,
        type: SAVE_STATUS
    }
}

export function setPlayedSeconds(playedSeconds: number): any{
    return {
        payload: playedSeconds,
        type: SET_PLAYED_SECONDS
    }
}

export const fetchMeta = (fileName: string, data: object) => (dispatch: any) => {
    Axios.put('/api/GetMeta?fileName=' + fileName, data)
        .then(meta => {
            dispatch({
                payload: meta,
                type: FETCH_META
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + FETCH_META))
        });
}
