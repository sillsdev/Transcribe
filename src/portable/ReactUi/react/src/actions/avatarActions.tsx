import { SAVE_AVATAR, SET_AVATAR_RADIUS, SET_SAVE_TO_PROJECT } from './types';

export function saveAvatar(avatar: IAvatar): any{
    return {
        payload: avatar,
        type: SAVE_AVATAR
    }
}

export function setUserAvatar(): any {
    return {
        payload: 100,
        type: SET_AVATAR_RADIUS
    }
}

export function setProjectAvatar(): any {
    return {
        payload: 1,
        type: SET_AVATAR_RADIUS
    }
}

export function setSaveToProject(projectId: string): any {
    return {
        payload: projectId,
        type: SET_SAVE_TO_PROJECT
    }
}