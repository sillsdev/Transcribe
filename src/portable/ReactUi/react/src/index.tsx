import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import AddManyTasks from './components/AddManyTasks';
import AvatarEdit from './components/AvatarEdit';
import MainLayout from './components/MainLayout';
import NewOrBrowseParatextProjects from './components/NewOrBrowseParatextProject';
import ProjectSettings from './components/ProjectSettings';
import SearchParatextProjects from './components/SearchParatextProjects';
import SelectProject from './components/SelectProject';
import TaskDetails from './components/TaskDetails';
import UiLang from './components/UiLang';
import UserDetails from './components/UserDetails';
import UserLogin from './components/UserLogin';
import UserSettings from './components/UserSettings';
import DevTools from './DevTool'
import './index.sass';
import registerServiceWorker from './registerServiceWorker';
import store from './store';

// https://reacttraining.com/react-router/web/example/basic
ReactDOM.render(
  <Provider store={store}>
    <Router>
      <div>
        <Route exact={true} path="/" component={UserLogin}/>
        <Route path="/project" component={SelectProject} />
        <Route path="/main" component={MainLayout} />
        <Route path="/main/avatar/Project" component={AvatarEdit} />
        <Route path="/settings" component={UserSettings}/>
        <Route path="/settings/avatar/User" component={AvatarEdit} />
        <Route path="/uilang" component={UiLang} />
        <Route path="/SearchParatextProjects" component={SearchParatextProjects} />
        <Route path="/NewOrBrowseParatextProjects" component={NewOrBrowseParatextProjects} />
        <Route path="/ProjectSettings" component={ProjectSettings} />
        <Route path="/ProjectSettings/Task" component={TaskDetails} />
        <Route path="/ProjectSettings/NewTask" component={TaskDetails} />
        <Route path="/ProjectSettings/User" component={UserDetails} />
        <Route path="/ProjectSettings/NewUser" component={UserDetails} />
        <Route path="/ProjectSettings/User/avatar/PopupUser" component={AvatarEdit} />
        <Route path="/ProjectSettings/NewUser/avatar/PopupUser" component={AvatarEdit} />
        <Route path="/ProjectSettings/AddManyTasks" component={AddManyTasks} />
        <Route path="/" component={DevTools} />
      </div>
    </Router>
  </Provider>,
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();
