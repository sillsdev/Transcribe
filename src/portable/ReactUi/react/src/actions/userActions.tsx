import Axios from 'axios';
import { log } from '../actions/logAction';
import { DELETE_USER, FETCH_USERS, RESTORE_DEFAULT_USER_HOTKEYS, SELECT_POPUP_USER, SELECT_USER, UPDATE_USER_PENDING } from './types';

export const fetchUsers = () => (dispatch: any) => {
    Axios.get('/api/GetUsers')
        .then(users => {
            dispatch({
                payload: users,
                type: FETCH_USERS
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + FETCH_USERS))
        });
}

export function selectUser(id: string): any{
    return {
        payload: id,
        type: SELECT_USER
    }
}

export function selectPopupUser(id: string): any{
    return {
        payload: id,
        type: SELECT_POPUP_USER
    }
}

export const selectLanguage = (user: string, lang: string) => (dispatch: any) => {
    dispatch({type: UPDATE_USER_PENDING});
    Axios.put('/api/UpdateUser?user=' + user + '&uilang=' + lang)
        .then(dispatch(fetchUsers()))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " select language, user=" + user + ", language=" + lang))
        });
}

export const updateUser = (user: string, project: string, query: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_USER_PENDING});
    Axios.put('/api/UpdateUser?user=' + user + '&project=' + project + query, data)
        .then(dispatch(fetchUsers()))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " update user, user=" + user))
        });
}

export const deleteUser = (user: string) => (dispatch: any) => {
    dispatch({type: DELETE_USER});
    Axios.put('/api/DeleteUser?user=' + user)
        .then(dispatch(fetchUsers()))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + DELETE_USER + ", user=" + user))
        });
}

export const saveUserSetting = (user: string, setting: string, value: string) => (dispatch: any) => {
    Axios.put('/api/UpdateUser?user=' + user + "&setting=" + setting + "&value=" + value)
    .catch((reason: any) => {
        dispatch(log(JSON.stringify(reason) + " save user setting, user=" + user+ ", setting=" + setting + ", value=" + value))
    });
}

export const restoreDefaultUserHotKeys = () => (dispatch: any) => {
    Axios.get('/api/GetDefaultUserHotKeys')
        .then(userHotKeys => {
            dispatch({
                payload: userHotKeys,
                type: RESTORE_DEFAULT_USER_HOTKEYS
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + RESTORE_DEFAULT_USER_HOTKEYS))
        });
}
