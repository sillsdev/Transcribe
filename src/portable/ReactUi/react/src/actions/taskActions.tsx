import Axios from 'axios';
import { log } from '../actions/logAction';
import { setSubmitted } from './audioActions';
import { ADD_MANY_TASKS, ASSIGN_TASK_PENDING, COMPLETE_REVIEW_PENDING,
    COMPLETE_TRANSCRIPTION_PENDING, COPY_TO_CLIPBOARD, DELETE_TASK,
    FETCH_FILTERED_TASK, FETCH_TASKS, FETCH_TRANSCRIPTION, FETCH_ZTT_PROJECTS_COUNT, INIT_TASKS, SELECT_POPUP_TASK,
    SELECT_PROJECT, SELECT_TASK, SET_SELECTED_OPTION, SET_TODO_HIGHLIGHT, SHOW_HELP, UNASSIGN_TASK_PENDING, UPDATE_PROJECT, UPDATE_PROJECT_AVATAR,  UPDATE_TASK,
    WRITE_FULFILLED, WRITE_PENDING } from './types';
import { fetchUsers, saveUserSetting } from './userActions';

export const assignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: ASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Assigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
        .then(dispatch(setSubmitted(false)))
        .then(dispatch(selectTask(userid, taskid)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + ASSIGN_TASK_PENDING +  ", id=" + taskid + ", user=" + userid))
        });
}

export const unAssignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: UNASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Unassigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(saveUserSetting(userid, "lastTask", "")))
        .then(dispatch(selectTask(userid, "")))
        .then(dispatch(fetchUsers()))
        .then(dispatch(fetchTasks(userid)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + UNASSIGN_TASK_PENDING +  ", id=" + taskid + ", user=" + userid))
        });
}

export const completeTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_TRANSCRIPTION_PENDING});
    Axios.put('/api/TaskEvent?action=TranscribeEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(selectTask(userid, "")))
        .then(dispatch(fetchUsers()))
        .then(dispatch(fetchTasks(userid)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + COMPLETE_TRANSCRIPTION_PENDING +  ", id=" + taskid + ", user=" + userid))
        });
}

export const completeReview = (taskid: string, userid: string, heading: string, sync: boolean | undefined) => (dispatch: any) => {
    dispatch({type: COMPLETE_REVIEW_PENDING});
    Axios.put('/api/TaskEvent?action=ReviewEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(sync? uploadTranscription(taskid, userid, heading): fetchTasks(userid)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + COMPLETE_REVIEW_PENDING +  ", id=" + taskid + ", user=" + userid))
        });
}

export const uploadTranscription = (taskid: string, userid: string, heading: string) => (dispatch: any) => {
    Axios.put('/api/TaskEvent?action=Upload&task=' + taskid + '&user=' + userid + '&heading=' + heading)
        .then(dispatch(selectTask(userid, "")))
        .then(dispatch(fetchUsers()))
        .then(dispatch(fetchTasks(userid)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " upload transcription, id=" + taskid + ", user=" + userid))
        });
}

export const writeTranscription = (taskid: string, length: number, lang: string, dir: string, data: any) => (dispatch: any) => {
    dispatch({type: WRITE_PENDING});
    Axios.put('/api/WriteTranscription?task=' + taskid + "&length=" + length.toString() + "&lang=" + lang + "&dir=" + dir, data)
        .then(dispatch({type: WRITE_FULFILLED}))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + WRITE_PENDING +  ", id=" + taskid))
        });
}

export const fetchTasks = (username: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?user=' + username)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch tasks, user=" + username))
        });
}

export const fetchTasksOfProject = (projectname: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?project=' + projectname)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
            dispatch({
                payload: projectname,
                type: SELECT_PROJECT
            })
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch tasks of project, id=" + projectname))
        });
    }

export const fetchZttProjectsCount = () => (dispatch: any) => {
    Axios.get('/api/GetZttProjectsCount')
        .then(zttProjectsCount => {
            dispatch({
                payload: zttProjectsCount,
                type: FETCH_ZTT_PROJECTS_COUNT
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch Ztt Projects Count"))
        });
}


export const deleteTask = (taskid: string) => (dispatch: any) => {
    const project = taskid.split('-')[0];
    dispatch({type: DELETE_TASK});
    Axios.put('/api/DeleteTask?action=Unassigned&task=' + taskid )
        .then(dispatch(fetchTasksOfProject(project)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + DELETE_TASK +  ", id=" + taskid))
        });
}

export function selectProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PROJECT
    }
}

export const selectTask = (user: string, id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_TASK
    })
    dispatch(fetchTasks(user))
    dispatch(fetchTranscription(id));
    dispatch(saveUserSetting(user, "lastTask", id));
    dispatch(setSubmitted(false));
}

export const  selectPopupTask = (id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_POPUP_TASK
    })
}

export const fetchTranscription = (taskid: string) => (dispatch: any) => {
    const part = taskid && taskid.split('.');
    if (part) {
        Axios.get('/api/audio/' + part[0] + '.transcription')
        .then(transcription => {
            dispatch({
                payload: transcription,
                type: FETCH_TRANSCRIPTION
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + FETCH_TRANSCRIPTION +  ", id=" + taskid))
        });
    }
}

export const updateTask = (task: string, project: string, query: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_TASK});
    Axios.put('/api/UpdateTask?task=' + task + '&project=' + project + query, data)
        .then(dispatch(fetchTasksOfProject(project)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + UPDATE_TASK +  ", id=" + task))
        });
}

export const updateProject = (project: IProject) => (dispatch: any) => {
    dispatch({
        payload: project.id,
        type: UPDATE_PROJECT
    })
    Axios.put('/api/UpdateProject?project=' + project.id + '&name=' + project.name + '&guid=' + project.guid + '&lang=' + project.lang + '&langName=' + project.langName + '&font=' + project.font + '&size=' + project.size + '&features=' + project.features + '&dir=' + project.direction + '&sync=' + project.sync + '&claim=' + project.claim + '&type=' + project.type + '&uri=' + project.uri )
        .then (dispatch(fetchTasksOfProject(project.id)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + UPDATE_PROJECT +  ", id=" + project.id))
        });
}

export const copyToClipboard = (taskid: string) => (dispatch: any) => {
    dispatch({
        payload: taskid,
        type: COPY_TO_CLIPBOARD
    })
    Axios.put('/api/CopyToClipboard?task=' + taskid)
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + COPY_TO_CLIPBOARD +  ", id=" + taskid))
        });
}


export const updateProjectAvatar = (user: string, project: string,  data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_PROJECT_AVATAR});
    Axios.put('/api/UpdateProjectAvatar?project=' + project, data)
        .then(dispatch(fetchTasks(user)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + UPDATE_PROJECT_AVATAR + ", user=" + user + ", project=" + project))
        });
}

export function initTasks(){
    return {
        type: INIT_TASKS
    }
}
export const addManyTasks = (user: string, project: string) => (dispatch: any) => {
    dispatch({type: ADD_MANY_TASKS});
    Axios.put('/api/AddManyTasks?user=' + user + '&project=' + project)
        .then(dispatch(fetchTasksOfProject(project)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + ADD_MANY_TASKS + ", project=" + project))
        });
}

export const showHelp = (topic: string) => (dispatch: any) => {
    dispatch({type: SHOW_HELP});
    Axios.put('/api/ShowHelp?topic=' + topic)
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + SHOW_HELP + ", topic=" + topic))
        });
}

export function setSelectedOption(selectedOption: string): any {
    return {
        payload: selectedOption,
        type: SET_SELECTED_OPTION
    }
}

export const fetchFilteredTasks = (username: string, project: string, option: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?user=' + username + '&project=' + project + '&option=' + option)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_FILTERED_TASK
            });
        })
        .then(dispatch(setSelectedOption(option)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + FETCH_FILTERED_TASK + ", user=" + username + ", project=" + project + ", option=" + option))
        });
}

export function setToDoHightlight(setTodoHighlight: boolean): any {
    return {
        payload: setTodoHighlight,
        type: SET_TODO_HIGHLIGHT
    }
}