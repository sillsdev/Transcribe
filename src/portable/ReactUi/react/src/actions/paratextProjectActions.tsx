import Axios from 'axios';
import { log } from '../actions/logAction';
import { fetchTasksOfProject } from './taskActions';
import { CLEAR_SELECTED_PARATEXT_PROJECT, FETCH_PARATEXT_PROJECTS, SELECT_PARATEXT_PROJECT } from './types';

export const fetchParatextProjects = () => (dispatch: any) => {
    Axios.get('/api/GetParatextProjects')
        .then(paratextProjects => {
            dispatch({
                payload: paratextProjects,
                type: FETCH_PARATEXT_PROJECTS
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch Paratext projects"))
        });
}

export const selectParatextProject = (project: IParatextProject) => (dispatch: any) => {
    dispatch({
        payload: project.id,
        type: SELECT_PARATEXT_PROJECT
    })
    Axios.put('/api/UpdateProject?project=' + project.id + '&name=' + project.name + '&guid=' + project.guid + '&lang=' + project.lang + '&langName=' + project.langName + '&font=' + project.font + '&size=' + project.size + '&features=' + project.features + '&dir=' + project.direction + '&type=' + project.type + '&uri=' + project.uri + '&sync=true&claim=true' )
        .then (dispatch(fetchTasksOfProject(project.id)))
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " " + SELECT_PARATEXT_PROJECT +  ", id=" + project.id))
        });
}

export function clearSelectedParatextProject(): any{
    return {
        payload: "",
        type: CLEAR_SELECTED_PARATEXT_PROJECT
    }
}