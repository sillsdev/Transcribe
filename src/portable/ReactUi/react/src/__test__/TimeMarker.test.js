/* jshint esversion: 6 */
import * as React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import * as actions from '../actions/audioActions';
import TimeMarker from '../components/controls/TimeMarker';

configure({ adapter: new Adapter() })

// Snapshot for TimeMarker
describe('>>>Control: TimeMarker --- Snapshot',()=>{
    const minProps = {
        playedSeconds: 10,
        totalSeconds: 30,
        timer: "countdown",
    };

    it('+++capturing Snapshot of TimeMarker', () => {
        const renderedValue =  renderer.create(<TimeMarker {...minProps}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: TimeMarker', () => {
    let wrapper;
    const minProps = {
        target: (context) => {return false;},
        text: "My Filter",
        type: "safe",
    };

    beforeEach(()=>{
        wrapper = shallow(<TimeMarker {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
