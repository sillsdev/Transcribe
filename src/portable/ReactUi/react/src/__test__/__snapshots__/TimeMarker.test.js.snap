// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: TimeMarker --- Snapshot +++capturing Snapshot of TimeMarker 1`] = `
<div
  className="TimeMarker"
  id="Time"
>
  <time
    dateTime="P20S"
    id="pos"
  >
    0:20
  </time>
  /
  <time
    dateTime="P30S"
    id="total"
  >
    0:30
  </time>
</div>
`;
